microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/DatabaseCrashTest.java

```diff
@@ -0,0 +1,211 @@
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.common.io.Files;
+import com.sun.security.auth.module.UnixSystem;
+
+/**
+ * Test MP Fault tolerance service with crashing database.
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class DatabaseCrashTest {
+
+    private static final String APPLICATION_NAME = "DatabaseServlet";
+
+    private static Docker postgresDB = null;
+    private static File postgresDataDir = null;
+
+    @ArquillianResource
+    private URL baseUrl;
+
+    @Deployment(testable = false)
+    public static WebArchive createDatabaseService() {
+        final WebArchive databaseService = ShrinkWrap.create(WebArchive.class, APPLICATION_NAME + ".war");
+        databaseService.addClasses(DatabaseService.class, DatabaseServlet.class);
+        databaseService.addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+
+        String postgresDS = "<datasources xmlns=\"http://www.jboss.org/ironjacamar/schema\">\n" +
+                "    <datasource jndi-name=\"java:jboss/datasources/PostgresDS\" pool-name=\"exampleDS\">\n" +
+                "        <driver>DatabaseServlet.war_org.postgresql.Driver_42_2</driver>\n" +
+                "        <connection-url>jdbc:postgresql://127.0.0.1:5432/test-database</connection-url>\n" +
+                "        <security>\n" +
+                "            <user-name>postgres</user-name>\n" +
+                "            <password>pass</password>\n" +
+                "        </security>\n" +
+                "           <validation>\n" +
+                "                <valid-connection-checker class-name=\"org.jboss.jca.adapters.jdbc.extensions.postgres.PostgreSQLValidConnectionChecker\">\n"
+                +
+                "                </valid-connection-checker>\n" +
+                "                <exception-sorter class-name=\"org.jboss.jca.adapters.jdbc.extensions.postgres.PostgreSQLExceptionSorter\">\n"
+                +
+                "                </exception-sorter>\n" +
+                "</validation>\n" +
+                "    </datasource>\n" +
+                "</datasources>";
+        databaseService.addAsWebInfResource(new StringAsset(postgresDS), "postgres-ds.xml");
+
+        // Add postgresql JDBC driver
+        JavaArchive[] postgresqlDriver = Maven.resolver().loadPomFromFile("pom.xml")
+                .resolve("org.postgresql:postgresql").withoutTransitivity().as(JavaArchive.class);
+        databaseService.addAsLibraries(postgresqlDriver);
+        return databaseService;
+    }
+
+    @BeforeClass
+    public static void startDatabase() throws Exception {
+        // create data dir for postgres database
+        postgresDataDir = Files.createTempDir();
```

Comment:
I would rather use `TemporaryFolder` rule which guarantees deletion upon completion of tests and you can skip caring of it in `@AfterClass`. Though this is just a FYI.

Reply:
Right, the reason why PostgreSQL docker container is not started in `@Rule` is because tests needs to operate with lifecycle of docker container (kill, restart again). I think that having rule just for creating folder is too much code.